cmd/api/routers/routers_test.go

```diff
@@ -18,28 +18,43 @@ import (
 )
 
 var testResources = fake.CreateTestResources()
-var testAPIResource = metav1.APIResource{
+var nonCoreResources = testResources[:1]
+var coreResources = testResources[1:]
+var testCoreAPIResource = metav1.APIResource{
+	Kind:         "Pod",
+	Name:         "pods",
+	Version:      "v1",
+	SingularName: "pod",
+	Namespaced:   true,
+}
+var testNonCoreAPIResource = metav1.APIResource{
 	Kind:         "Crontab",
 	Name:         "crontabs",
 	Group:        "stable.example.com",
 	Version:      "v1",
 	SingularName: "crontab",
 	Namespaced:   true}
 
-func setupRouter() *gin.Engine {
+func setupRouter(core bool) *gin.Engine {
 	router := gin.Default()
 	ctrl := Controller{Database: fake.NewFakeDatabase(testResources)}
 	router.Use(func(c *gin.Context) {
-		c.Set("apiResource", testAPIResource)
-		c.Next()
+		if core {
+			c.Set("apiResource", testCoreAPIResource)
+		} else {
+			c.Set("apiResource", testNonCoreAPIResource)
+			c.Next()
+		}
```

Comment:
```suggestion
		if core {
			c.Set("apiResource", testCoreAPIResource)
		} else {
			c.Set("apiResource", testNonCoreAPIResource)
		}
		c.Next()
```

Reply:
From the Next docs:

```
// Next should be used only inside middleware.
// It executes the pending handlers in the chain inside the calling handler.
// See example in GitHub.
```

And from this example https://github.com/gin-gonic/gin/blob/master/logger.go#L249, is useful only if called before some code in the middleware. Is not useful to call it at the end. See also this issue: https://github.com/gin-gonic/gin/issues/287. So I'm editing my suggestion and we should review our current middleware at some point to make proper use. Its not that is not working, but it leads to some confusion, at least for me.

Reply:
Good research! 
I created https://github.com/kubearchive/kubearchive/issues/363 for addressing that thing in the middleware functions and I will remove the `next()` call in this test

Reply:
addressed